modules/navier_stokes/src/bcs/INSADDisplaceBoundaryBC.C

```diff
@@ -39,5 +45,6 @@ INSADDisplaceBoundaryBC::INSADDisplaceBoundaryBC(const InputParameters & paramet
 ADReal
 INSADDisplaceBoundaryBC::computeQpResidual()
 {
-  return _u - (_u_old + this->_dt * _velocity(_component));
+  const Moose::NodeArg nd{_current_node, _sub_id};
+  return _u - (_u_old + this->_dt * _velocity(nd, determineState())(_component));
```

Comment:
hardcoding Euler

lets have some check in the constructor?

Reply:
It's already there 10 lines up per your suggestion on the original PR that added this object :-)